### PR TITLE
Move Proposals listeners to Wallet Package

### PR DESCRIPTION
**Pull request by Sirmorrison**
* Move proposals initialization from proposals page to state.go
* Move Proposals sync listeners to wallet package.
* Restored the restore button after sync is completed
* Syncs proposal when no proposal has been fetched (initial sync).

### CODE REVIEW

File: wallet/commands.go [hunk 1]
```diff
@@ -653,17 +653,30 @@ func (wal *Wallet) RenameAccount(walletID int, acct int32, name string, errChan
...
 }
 
+func (wal *Wallet) ClearSavedProposals() {
+	err := wal.multi.Politeia.ClearSavedProposals()
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+}
+
```
Comment (oshorefueled) [inline wallet/commands.go]: This doesn't seem to be used anywhere

File: ui/proposals_page.go [hunk 2]
```diff
@@ -72,27 +47,25 @@ type tabs struct {
...
-	updatedIcon                *widget.Icon
-	updatedLabel               decredmaterial.Label
-	syncButton                 *widget.Clickable
-	startSyncIcon              *widget.Image
-	timerIcon                  *widget.Image
+	theme            *decredmaterial.Theme
+	wallet           *wallet.Wallet
+	selectedProposal **dcrlibwallet.Proposal
+	proposals        **wallet.Proposals
+	proposalSync     *chan *wallet.NewProposal
```
Comment (oshorefueled) [inline ui/proposals_page.go]: Is there any reason a pointer to the channel was used here?

File: ui/proposals_page.go [hunk 3]
```diff
@@ -168,81 +142,33 @@ func (pg *proposalsPage) Handle(common pageCommon) {
...
-func (pg *proposalsPage) onFetchError(err error) {
-	pg.state = categoryStateError
-	pg.onFetchComplete()
-	pg.notify(err.Error(), false)
-}
-
-func (pg *proposalsPage) onFetchComplete() {
-	if !pg.hasFetchedInitialProposals {
-		pg.hasFetchedInitialProposals = true
+	default:
```
Comment (oshorefueled) [inline ui/proposals_page.go:161]: There's no need for a default case if the case is empty

File: ui/proposals_page.go [hunk 4]
```diff
@@ -541,24 +496,45 @@ func (pg *proposalsPage) layoutStartSyncSection(gtx C) D {
...
+			}
+			for k := range pg.tabs.tabs {
+				if pg.tabs.tabs[k].category == (*pg.proposals).Proposals[i].Category {
+					pg.tabs.tabs[k].proposals = append(pg.tabs.tabs[k].proposals, item)
+					break
+				}
+			}
+		}
+	}
+}
```
Comment (oshorefueled) [inline ui/proposals_page.go]: This looks like a duplicate of the code in `addDiscoveredProposal` and the method can simply be called here.

File: ui/proposals_page.go [hunk 5]
```diff
@@ -308,23 +234,56 @@ func (pg *proposalsPage) layoutTabs(gtx C) D {
...
+			}
+			pg.tabs.tabs[i].proposals = append([]proposalItem{item}, pg.tabs.tabs[i].proposals...)
+			break
+		}
+	}
 }
 
-func (pg *proposalsPage) layoutErrorState(gtx C) D {
-	str := "Error fetching proposals"
+func (pg *proposalsPage) updateProposal(proposal dcrlibwallet.Proposal) {
```
Comment (oshorefueled) [inline ui/proposals_page.go]: Kindly add a comment here or a more descriptive method name. What about the proposal is being updated?

File: wallet/proposal_listener.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+type ProposalStatus int
+
+const (
+	Synced ProposalStatus = iota
+	VoteStarted
+	NewProposalFound
+	VoteFinished
+)
+
+type NewProposal struct {
```
Comment (oshorefueled) [inline wallet/proposal_listener.go]: Proposal is just fine here

**Review by oshorefueled — CHANGES_REQUESTED** [review]

File: ui/proposals_page.go (hunk 2 above)
Comment (Sirmorrison) [reply ui/proposals_page.go]: It is returning the proposal sync status and also the proposal from the proposal listener.

File: ui/proposals_page.go (hunk 3 above)
Comment (Sirmorrison) [reply ui/proposals_page.go:161]: Removing the default cases causes the app to freeze when proposals page is selected. I spent the last 30 mins trying to figure out this behavior.

File: ui/proposals_page.go (hunk 2 above)
Comment (oshorefueled) [reply ui/proposals_page.go]: I don't think you need a pointer to a channel to do that

File: ui/proposals_page.go (hunk 3 above)
Comment (oshorefueled) [reply ui/proposals_page.go:161]: oh I see. It's a select case and removing the default case causes the app to wait for a signal from the syncedProposal channel.

**Review by oshorefueled — CHANGES_REQUESTED** [review]

**Review by oshorefueled — APPROVED** [review]

**Pull request merged**